src/accumulator.c

```diff
@@ -0,0 +1,208 @@
+	pg_brick_get_state(brick, struct pg_accumulator_state);
+	struct pg_brick_side *s =
+	&brick->sides[state->output];
+	uint64_t pkts_mask;
+	bool ret = false;
+	enum pg_side poll_dir = pg_flip_side(state->output);
+
+	if (state->output == PG_MAX_SIDE) {
+		s = &brick->sides[PG_WEST_SIDE];
+	} else if (state->poll_count[poll_dir] < pg_poll_min) {
+		++state->poll_count[poll_dir];
+		return 0;
+	}
+	if (unlikely(s->edge.link == NULL))
+		return 0;
+	if (state->output == PG_MAX_SIDE) {
+		if (state->poll_count[PG_WEST_SIDE] < pg_poll_min) {
+			++state->poll_count[PG_WEST_SIDE];
+			ret = true;
+		} else if (state->poll_count[PG_EAST_SIDE] < pg_poll_min) {
+			++state->poll_count[PG_EAST_SIDE];
+			ret = true;
+		}
+		if (ret)
+			return 0;
+			for (int i = 0; i < 2; i++) {
+				s = &brick->sides[i];
+				pkts_mask = (1 << state->burst_vec.len[i]) - 1;
+				*pkts_cnt = state->burst_vec.len[i];
+				pg_brick_burst(s->edge.link,
+					       i, s->edge.pair_index,
+					       state->burst_vec.mbufs[i],
+					       pkts_mask, errp);
+				pg_packets_free(state->burst_vec.mbufs[i],
+						pkts_mask);
+				state->burst_vec.len[i] = 0;
+
+		}
+	} else {
+		pkts_mask = (1 << state->burst_vec.len[poll_dir]) - 1;
+		*pkts_cnt = state->burst_vec.len[poll_dir];
+		pg_brick_burst(s->edge.link,
+			       state->output, s->edge.pair_index,
+			       state->burst_vec.mbufs[poll_dir],
+			       pkts_mask, errp);
+		pg_packets_free(state->burst_vec.mbufs[poll_dir], pkts_mask);
+		state->burst_vec.len[poll_dir] = 0;
+	}
+	state->poll_count[PG_WEST_SIDE] = 0;
+	state->poll_count[PG_EAST_SIDE] = 0;
+
+	return 0;
+}
+
+static int accumulator_init(struct pg_brick *brick,
+			    struct pg_brick_config *config,
+			    struct pg_error **errp)
+{
+	struct pg_accumulator_state *state;
+	struct pg_accumulator_config *accumulator_config;
+
+	state = pg_brick_get_state(brick, struct pg_accumulator_state);
+
+	accumulator_config =
+	(struct pg_accumulator_config *) config->brick_config;
+
+	brick->burst = accumulator_burst;
+	brick->poll = accumulator_poll;
+	state->output = accumulator_config->output;
+	state->burst_vec = pg_vecnew();
+	state->poll_count[PG_WEST_SIDE] = 0;
+	state->poll_count[PG_EAST_SIDE] = 0;
+
+	return 0;
+}
+
+static struct pg_brick_config *accumulator_config_new(const char *name,
+						      enum pg_side output)
+{
+	struct pg_brick_config *config = g_new0(struct pg_brick_config, 1);
+	struct pg_accumulator_config *accumulator_config =
+		g_new0(struct pg_accumulator_config, 1);
+
+	accumulator_config->output = output;
+	config->brick_config = (void *) accumulator_config;
+	return pg_brick_config_init(config, name, 1, 1, PG_DIPOLE);
+}
+
+struct pg_brick *pg_accumulator_new(const char *name,
+				    enum pg_side output,
+				    uint64_t poll_min,
+				    struct pg_error **errp)
+{
+	struct pg_brick_config *config = accumulator_config_new(name, output);
+	struct pg_brick *ret = pg_brick_new("accumulator", config, errp);
+
+	pg_poll_min = poll_min;
```

Comment:
so at each new brick you change a global use on each brick ?
that doesn't seems right